### ISSUE TITLE
sphinx directives: AttributeError: Values instance has no attribute 'env'

### ISSUE THREAD

**Issue by moylop260** [twolfson/restructuredtext-lint#29]
Steps:
 - Install:
  - `virtualenv test && source test/bin/activate && pip install restructuredtext_lint sphinx`
 - Test:
  - `python -c "import restructuredtext_lint;from sphinx.directives import other;restructuredtext_lint.lint('.. toctree::\n   :maxdepth: 1\n\n   project/contribute')"`

Saw:
```bash
Traceback (most recent call last):
  File "<string>", line 1, in <module>
  File "restructuredtext_lint/lint.py", line 62, in lint
    reader.parser.parse(content, document)
  File "docutils/parsers/rst/__init__.py", line 172, in parse
    self.statemachine.run(inputlines, document, inliner=self.inliner)
  File "docutils/parsers/rst/states.py", line 170, in run
    input_source=document['source'])
  File "docutils/statemachine.py", line 239, in run
    context, state, transitions)
  File "docutils/statemachine.py", line 460, in check_line
    return method(match, context, next_state)
  File "docutils/parsers/rst/states.py", line 2299, in explicit_markup
    nodelist, blank_finish = self.explicit_construct(match)
  File "docutils/parsers/rst/states.py", line 2311, in explicit_construct
    return method(self, expmatch)
  File "docutils/parsers/rst/states.py", line 2054, in directive
    directive_class, match, type_name, option_presets)
  File "docutils/parsers/rst/states.py", line 2103, in run_directive
    result = directive_instance.run()
  File "sphinx/directives/other.py", line 52, in run
    env = self.state.document.settings.env
AttributeError: Values instance has no attribute 'env'
```

 - Environment

```bash
$pip freeze
alabaster==0.7.8
Babel==2.3.4
docutils==0.12
imagesize==0.7.1
Jinja2==2.8
MarkupSafe==0.23
Pygments==2.1.3
pytz==2016.4
restructuredtext-lint==0.14.3
six==1.10.0
snowballstemmer==1.2.1
Sphinx==1.4.4
```

`$python --version: Python 2.7.6`

**Comment by moylop260** [comment]
Is this a `restructuredtext-lint` issue or should be a `sphinx` issue?

**Comment by twolfson** [comment]
What happens when you don't import Sphinx?

**Comment by moylop260** [comment]
> @twolfson
What happens when you don't import Sphinx?

`python -c "import restructuredtext_lint;print [m.message for m in restructuredtext_lint.lint('.. toctree::\n   :maxdepth: 1\n\n   project/contribute')]"`
`[u'No directive entry for "toctree" in module "docutils.parsers.rst.languages.en".\nTrying "toctree" as canonical directive name.', u'Unknown directive type "toctree".']`

**Comment by twolfson** [comment]
Ah, damn. Nothing helpful =( I have a feeling that it's a problem with not fully configuring Sphinx but am not familiar enough with Sphinx.

As noted in the README, we have no errors if we register a simple Sphinx directive like `highlight`:

https://github.com/twolfson/restructuredtext-lint/tree/0.14.3#extension

**Comment by moylop260** [comment]
@twolfson 
FYI https://github.com/sphinx-doc/sphinx/issues/2922#issuecomment-243377010

**Comment by twolfson** [comment]
Ah, sounds like there should be another repo for handling that then (e.g. `sphinx-lint`). Thoughts on closing this issue as invalid and updating the README content with a link to that issue?

**Comment by moylop260** [comment]
+1

**Comment by twolfson** [comment]
The updated docs have been published in `0.17.1`. Thanks for being so proactive on this =)
